lua/wire/client/cl_wirelib.lua

```diff
@@ -162,35 +162,42 @@ function Wire_DrawTracerBeam( ent, beam_num, hilight, beam_length )
 	if (beam_length > 0) then
 
 		local x, y = 0, 0
-		if (ent.GetSkewX and ent.GetSkewY) then
+		if ent.GetTarget and ( ent:GetTarget().X ~= 0 or ent:GetTarget().Y ~= 0 or ent:GetTarget().Z ~= 0 ) then
+			ent.endpos = ent:GetPos() + ( ent:GetTarget() - ent:GetPos() ):GetNormalized()*beam_length
+			if tracedata.endpos[1] ~= tracedata.endpos[1] then tracedata.endpos = self:GetPos()+Vector(self:GetBeamLength(), 0, 0) end
+		elseif (ent.GetSkewX and ent.GetSkewY) then
 			x, y = ent:GetSkewX(beam_num), ent:GetSkewY(beam_num)
-		end
-
-		local start, ang = ent:GetPos(), ent:GetAngles()
-
-		if (ent.ls != start or ent.la != ang or ent.ll != beam_length or ent.lx != x or ent.ly != y) then
-			ent.ls, ent.la = start, ang
-
-			if (ent.ll != beam_length or ent.lx != x or ent.ly != y) then
-				ent.ll, ent.lx, ent.ly = beam_length, x, y
-
-				if (x == 0 and y == 0) then
-					ent.endpos = start + (ent:GetUp() * beam_length)
+			
+			if (ent.ls ~= start or ent.la ~= ang or ent.ll ~= beam_length or ent.lx ~= x or ent.ly ~= y) then
+				ent.ls, ent.la = start, ang
+
+				if (ent.ll ~= beam_length or ent.lx ~= x or ent.ly ~= y) then
+					ent.ll, ent.lx, ent.ly = beam_length, x, y
+
+					if (x == 0 and y == 0) then
+						ent.endpos = start + (ent:GetUp() * beam_length)
+					else
+						local skew = Vector(x, y, 1)
+						skew = skew*(beam_length/skew:Length())
+						local beam_x = ent:GetRight()*skew.x
+						local beam_y = ent:GetForward()*skew.y
+						local beam_z = ent:GetUp()*skew.z
+						ent.endpos = start + beam_x + beam_y + beam_z
+					end
+					ent.ExtraRBoxPoints = ent.ExtraRBoxPoints or {}
+					ent.ExtraRBoxPoints[beam_num] = ent:WorldToLocal(ent.endpos)
 				else
-					local skew = Vector(x, y, 1)
-					skew = skew*(beam_length/skew:Length())
-					local beam_x = ent:GetRight()*skew.x
-					local beam_y = ent:GetForward()*skew.y
-					local beam_z = ent:GetUp()*skew.z
-					ent.endpos = start + beam_x + beam_y + beam_z
+					ent.endpos = ent:LocalToWorld(ent.ExtraRBoxPoints[beam_num])
 				end
-				ent.ExtraRBoxPoints = ent.ExtraRBoxPoints or {}
-				ent.ExtraRBoxPoints[beam_num] = ent:WorldToLocal(ent.endpos)
-			else
-				ent.endpos = ent:LocalToWorld(ent.ExtraRBoxPoints[beam_num])
 			end
+		else
+			ent.endpos = ent:GetPos() + ent:GetUp()*beam_length
 		end
 
+		local start, ang = ent:GetPos(), ent:GetAngles()
```

Comment:
Moving this down here causes these vars to be undefined in the above code. Can you also rename 'start' to 'pos' and replace all occurrences of ent:GetPos() with pos?